extensions/dicom-seg/src/components/SegmentationPanel.js

```diff
@@ -0,0 +1,473 @@
+
+    return () => {
+      cornerstoneTools.store.state.enabledElements.forEach(enabledElement =>
+        enabledElement.removeEventListener(
+          'cornersontetoolslabelmapmodified',
+          labelmapModifiedHandler
+        )
+      );
+
+      document.removeEventListener('side-panel-change', handleSidePanelChange);
+    };
+  });
+
+  const handleSidePanelChange = () => {
+    setShowSegSettings(false);
+  };
+
+  if (!brushStackState) {
+    return null;
+  }
+
+  const labelmap3D =
+    brushStackState.labelmaps3D[brushStackState.activeLabelmapIndex];
+
+  /*
+   * 2. UseEffect to update state? or to a least trigger a re-render
+   * 4. Toggle visibility of labelmap?
+   * 5. Toggle visibility of seg?
+   *
+   * If the port is cornerstone, just need to call a re-render.
+   * If the port is vtkjs, its a bit more tricky as we now need to create a new
+   */
+
+  const getLabelmapList = () => {
+    /* Get list of SEG labelmaps specific to active viewport (reference series) */
+    const referencedSegDisplaysets = _getReferencedSegDisplaysets(
+      studyInstanceUid,
+      seriesInstanceUid
+    );
+
+    return referencedSegDisplaysets.map((displaySet, index) => {
+      const { labelmapIndex, seriesDate, seriesTime } = displaySet;
+
+      /* Map to display representation */
+      const dateStr = `${seriesDate}:${seriesTime}`.split('.')[0];
+      const date = moment(dateStr, 'YYYYMMDD:HHmmss');
+      const isActiveLabelmap =
+        labelmapIndex === brushStackState.activeLabelmapIndex;
+      const displayDate = date.format('ddd, MMM Do YYYY');
+      const displayTime = date.format('h:mm:ss a');
+      const displayDescription = displaySet.seriesDescription;
+
+      return {
+        value: labelmapIndex,
+        title: displayDescription,
+        description: displayDate,
+        onClick: async () => {
+          const activatedLabelmapIndex = await _setActiveLabelmap(
+            viewport,
+            studies,
+            displaySet,
+            firstImageId,
+            brushStackState.activeLabelmapIndex
+          );
+          setSelectedSegmentation(activatedLabelmapIndex);
+        },
+      };
+    });
+  };
+
+  const labelmapList = getLabelmapList();
+
+  const segmentList = [];
+
+  if (labelmap3D) {
+    /*
+     * Newly created segments have no `meta`
+     * So we instead build a list of all segment indexes in use
+     * Then find any associated metadata
+     */
+    const uniqueSegmentIndexes = labelmap3D.labelmaps2D
+      .reduce((acc, labelmap2D) => {
+        if (labelmap2D) {
+          const segmentIndexes = labelmap2D.segmentsOnLabelmap;
+
+          for (let i = 0; i < segmentIndexes.length; i++) {
+            if (!acc.includes(segmentIndexes[i]) && segmentIndexes[i] !== 0) {
+              acc.push(segmentIndexes[i]);
+            }
+          }
+        }
+
+        return acc;
+      }, [])
+      .sort((a, b) => a - b);
```

Comment:
This is kind of what the `metadata` field of the `labelmap3D` is for. However its optional as any data could be put in there, and the brushes don't fill it by default.

Either we make it so that the default brush tools add a default (blank object?) metadata entry when a segment is drawn, or we could add this as a helper in CST?

Reply:
@dannyrb pending?